go/channelling/room_manager.go

```diff
@@ -209,9 +209,27 @@ func (rooms *roomManager) Get(roomID string) (room RoomWorker, ok bool) {
 	return
 }
 
+func (rooms *roomManager) isPublicRoom(roomName string) bool {
+	if rooms.PublicRoomNames != nil &&
+		rooms.PublicRoomNames.MatchString(roomName) {
+		return true
+	}
+	return false
+}
+
 func (rooms *roomManager) GetOrCreate(roomID, roomName, roomType string, credentials *DataRoomCredentials, sessionAuthenticated bool) (RoomWorker, error) {
+	isPublic := -1 // Will be evaluated lazily
```

Comment:
> you are evaluating the regular expression twice

No, see this check here: https://github.com/strukturag/spreed-webrtc/pull/436/files#r132847105
When `isPublic` is `false` there, the regex was not used yet, as we bail out here: https://github.com/strukturag/spreed-webrtc/pull/436/files#r132847052